snakemake/deployment/conda.py

```diff
@@ -522,7 +522,9 @@ def create(self, dryrun=False):
 
     @classmethod
     def get_singularity_envvars(self):
-        return {"CONDA_PKGS_DIRS": "/tmp/conda/{}".format(uuid.uuid4())}
+        return {
+            "CONDA_PKGS_DIRS": "{}/conda/{}".format(tempfile.gettempdir(), uuid.uuid4())
```

Comment:
Maybe my (testing) setup could be responsible? I'm pretty new to docker/singularity.

I was running snakemake installed through conda outside any containers. And that snakemake call was invoked with `--singularity --use_conda`. While installing envs I was getting some storage issues. From memory I think the env var was actually `SINGULARITYENV_CONDA_PKGS_DIRS`. But when I changed this in source the storage issues disappeared. 

Let me re-run to double check if it wasn't something else

Reply:
If I fill the tmpdir I'm instantly getting:

```
CreateCondaEnvironmentException:
Could not create conda environment from /vol/mbconda/sande/seq2science/lib/python3.8/site-packages/seq2science/rules/../envs/fastp.yaml:
Command:
SINGULARITYENV_CONDA_PKGS_DIRS=/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661 singularity  exec --home /scratch/sande/singularity_test  /scratch/sande/singularity_test/.snakemake/singularity/0eb6f8016568e58b92f11a8a123a998d.simg sh -c 'mamba env create --quiet --file "/vol/mbconda/sande/seq2science/lib/python3.8/site-packages/seq2science/.snakemake/d57f534c6fc4b3e4384e7bb6a7431d07.yaml" --prefix "/vol/mbconda/sande/seq2science/lib/python3.8/site-packages/seq2science/.snakemake/d57f534c6fc4b3e4384e7bb6a7431d07
```

I'm calling snakemake from the api, but I'm using `use_conda` and `use_singularity`. Without singularity (just conda) I'm not having issues. It doesn't make a difference whether or not I set the conda_prefix.

Reply:
Is there no further error message (what exactly did singularity complain about)?

Reply:
I think the output is captured by snakemake. If I run it in the shell I get:

```
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache' failed, continuing without it
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache' failed, continuing without it
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache' failed, continuing without it
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache' failed, continuing without it
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache' failed, continuing without it
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache' failed, continuing without it
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache' failed, continuing without it
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache' failed, continuing without it
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache/b63425f9.json'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661/cache/b63425f9.json'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/usr/local/lib/python3.8/site-packages/conda/exceptions.py", line 1080, in __call__
        return func(*args, **kwargs)
      File "/usr/local/lib/python3.8/site-packages/conda_env/cli/main.py", line 80, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/usr/local/lib/python3.8/site-packages/conda_env/cli/main_create.py", line 141, in execute
        result[installer_type] = installer.install(prefix, pkg_specs, args, env)
      File "/usr/local/lib/python3.8/site-packages/mamba/mamba_env.py", line 44, in mamba_install
        index = load_channels(pool, channel_urls, repos, prepend=False)
      File "/usr/local/lib/python3.8/site-packages/mamba/utils.py", line 122, in load_channels
        index = get_index(
      File "/usr/local/lib/python3.8/site-packages/mamba/utils.py", line 103, in get_index
        is_downloaded = dlist.download(True)
    RuntimeError: LockFile error. Aborting.

`$ /usr/local/bin/mamba create --quiet --file /vol/mbconda/sande/seq2science/lib/python3.8/site-packages/seq2science/.snakemake/d57f534c6fc4b3e4384e7bb6a7431d07.yaml --prefix /vol/mbconda/sande/seq2science/lib/python3.8/site-packages/seq2science/.snakemake/d57f534c6fc4b3e4384e7bb6a7431d07`

  environment variables:
                 CIO_TEST=<not set>
  CONDA_AUTO_UPDATE_CONDA=false
        CONDA_DEFAULT_ENV=/vol/mbconda/sande/seq2science
                CONDA_EXE=/vol/mbconda/miniconda3/bin/conda
          CONDA_PKGS_DIRS=/tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661
             CONDA_PREFIX=/vol/mbconda/sande/seq2science
           CONDA_PREFIX_1=/vol/mbconda/miniconda3
    CONDA_PROMPT_MODIFIER=(seq2science)
         CONDA_PYTHON_EXE=/vol/mbconda/miniconda3/bin/python
               CONDA_ROOT=/usr/local
              CONDA_SHLVL=2
           CURL_CA_BUNDLE=<not set>
          LD_LIBRARY_PATH=/.singularity.d/libs
                  MANPATH=/usr/local/slurm/share/man:/usr/local/openmpi/share/man:
               MODULEPATH=/etc/environment-modules/modules:/usr/share/modules/versions:/usr/shar
                          e/modules/$MODULE_VERSION/modulefiles:/usr/share/modules/modulefiles
                     PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>
                USER_PATH=/vol/mbconda/sande/seq2science/bin:/vol/mbconda/miniconda3/condabin:/u
                          sr/local/slurm/bin:/usr/local/openmpi/bin:/usr/local/hwloc/bin:/usr/lo
                          cal/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/
                          local/cuda/bin:/opt/dell/srvadmin/bin:/home/sande/.aspera/connect/bin:
                          /bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin

     active environment : /vol/mbconda/sande/seq2science
    active env location : /vol/mbconda/sande/seq2science
            shell level : 2
       user config file : /scratch/sande/singularity_test/.condarc
 populated config files : 
          conda version : 4.11.0
    conda-build version : not installed
         python version : 3.8.12.final.0
       virtual packages : __linux=5.13.0=0
                          __glibc=2.28=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /usr/local  (read only)
      conda av data dir : /usr/local/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /tmp/conda/7bda119b-bea5-442c-8fd8-27cfa2095661
       envs directories : /scratch/sande/singularity_test/.conda/envs
                          /usr/local/envs
               platform : linux-64
             user-agent : conda/4.11.0 requests/2.25.1 CPython/3.8.12 Linux/5.13.0-35-generic debian/10 glibc/2.28
                UID:GID : 30800:30800
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.
```

Reply:
Never mind, snakemake does log some output. Seems to be different from when I run it, and also depending on where in the process it crashes (e.g. if storage is 100% full or e.g. 99.9%). 

```
CreateCondaEnvironmentException:
Could not create conda environment from /vol/mbconda/sande/seq2science/lib/python3.8/site-packages/seq2science/rules/../envs/dupradar.yaml:
Command:
SINGULARITYENV_CONDA_PKGS_DIRS=/tmp/conda/2f1f656d-1af2-4e1f-b21b-50ed1e038f6b singularity  exec --home /scratch/sande/singularity_test  /scratch/sande/singularity_test/.snakemake/singularity/0eb6f8016568e58b92f11a8a123a998d.simg sh -c 'mamba env create --quiet --file "/scratch/sande/singularity_test/envs/f5ec87b90667167dd21cfe01fbf6e664.yaml" --prefix "/scratch/sande/singularity_test/envs/f5ec87b90667167dd21cfe01fbf6e664"'
Output:
error    Could not find any writable cache directory for repodata file

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/usr/local/lib/python3.8/site-packages/conda/exceptions.py", line 1080, in __call__
        return func(*args, **kwargs)
      File "/usr/local/lib/python3.8/site-packages/conda_env/cli/main.py", line 80, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/usr/local/lib/python3.8/site-packages/conda_env/cli/main_create.py", line 141, in execute
        result[installer_type] = installer.install(prefix, pkg_specs, args, env)
      File "/usr/local/lib/python3.8/site-packages/mamba/mamba_env.py", line 44, in mamba_install
        index = load_channels(pool, channel_urls, repos, prepend=False)
      File "/usr/local/lib/python3.8/site-packages/mamba/utils.py", line 122, in load_channels
        index = get_index(
      File "/usr/local/lib/python3.8/site-packages/mamba/utils.py", line 103, in get_index
        is_downloaded = dlist.download(True)
    RuntimeError: Non-writable cache error.

`$ /usr/local/bin/mamba create --quiet --file /scratch/sande/singularity_test/envs/f5ec87b90667167dd21cfe01fbf6e664.yaml --prefix /scratch/sande/singularity_test/envs/f5ec87b90667167dd21cfe01fbf6e664`

  environment variables:
                 CIO_TEST=<not set>
  CONDA_AUTO_UPDATE_CONDA=false
        CONDA_DEFAULT_ENV=/vol/mbconda/sande/seq2science
                CONDA_EXE=/vol/mbconda/miniconda3/bin/conda
          CONDA_PKGS_DIRS=/tmp/conda/2f1f656d-1af2-4e1f-b21b-50ed1e038f6b
             CONDA_PREFIX=/vol/mbconda/sande/seq2science
           CONDA_PREFIX_1=/vol/mbconda/miniconda3
    CONDA_PROMPT_MODIFIER=(seq2science)
         CONDA_PYTHON_EXE=/vol/mbconda/miniconda3/bin/python
               CONDA_ROOT=/usr/local
              CONDA_SHLVL=2
           CURL_CA_BUNDLE=<not set>
          LD_LIBRARY_PATH=/.singularity.d/libs
                  MANPATH=/usr/local/slurm/share/man:/usr/local/openmpi/share/man:
               MODULEPATH=/etc/environment-modules/modules:/usr/share/modules/versions:/usr/shar
                          e/modules/$MODULE_VERSION/modulefiles:/usr/share/modules/modulefiles
                     PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>
                USER_PATH=/vol/mbconda/sande/seq2science/bin:/vol/mbconda/miniconda3/condabin:/u
                          sr/local/slurm/bin:/usr/local/openmpi/bin:/usr/local/hwloc/bin:/usr/lo
                          cal/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/
                          local/cuda/bin:/opt/dell/srvadmin/bin:/home/sande/.aspera/connect/bin:
                          /bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin

     active environment : /vol/mbconda/sande/seq2science
    active env location : /vol/mbconda/sande/seq2science
            shell level : 2
       user config file : /scratch/sande/singularity_test/.condarc
 populated config files : 
          conda version : 4.11.0
    conda-build version : not installed
         python version : 3.8.12.final.0
       virtual packages : __linux=5.13.0=0
                          __glibc=2.28=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /usr/local  (read only)
      conda av data dir : /usr/local/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /tmp/conda/2f1f656d-1af2-4e1f-b21b-50ed1e038f6b
       envs directories : /scratch/sande/singularity_test/.conda/envs
                          /usr/local/envs
               platform : linux-64
             user-agent : conda/4.11.0 requests/2.25.1 CPython/3.8.12 Linux/5.13.0-35-generic debian/10 glibc/2.28
                UID:GID : 30800:30800
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.
```

A different error which occured when not fully full

```
CreateCondaEnvironmentException:
Could not create conda environment from /vol/mbconda/sande/seq2science/lib/python3.8/site-packages/seq2science/rules/../envs/htmltable.yaml:
Command:
SINGULARITYENV_CONDA_PKGS_DIRS=/tmp/conda/349dcbf7-2a57-4fbb-9e11-ef6d4470f655 singularity  exec --home /scratch/sande/singularity_test  /scratch/sande/singularity_test/.snakemake/singularity/0eb6f8016568e58b92f11a8a123a998d.simg sh -c 'mamba env create --quiet --file "/scratch/sande/singularity_test/envs/d8283b9fd7b570b6dae90b7c24328173.yaml" --prefix "/scratch/sande/singularity_test/envs/d8283b9fd7b570b6dae90b7c24328173"'
Output:
  Package                Version  Build                Channel                     Size
─────────────────────────────────────────────────────────────────────────────────────────
  Install:
─────────────────────────────────────────────────────────────────────────────────────────

  + _libgcc_mutex            0.1  conda_forge          conda-forge/linux-64        3 KB
  + _openmp_mutex            4.5  1_gnu                conda-forge/linux-64       22 KB
  + bzip2                  1.0.8  h7f98852_4           conda-forge/linux-64      484 KB
  + ca-certificates    2021.10.8  ha878542_0           conda-forge/linux-64      139 KB
  + ld_impl_linux-64      2.36.1  hea4e1c9_2           conda-forge/linux-64      667 KB
  + libblas                3.9.0  14_linux64_openblas  conda-forge/linux-64       12 KB
  + libcblas               3.9.0  14_linux64_openblas  conda-forge/linux-64       12 KB
  + libffi                 3.4.2  h7f98852_5           conda-forge/linux-64       57 KB
  + libgcc-ng             11.2.0  h1d223b6_15          conda-forge/linux-64      911 KB
  + libgfortran-ng        11.2.0  h69a702a_15          conda-forge/linux-64       23 KB
  + libgfortran5          11.2.0  h5c6108e_15          conda-forge/linux-64        2 MB
  + libgomp               11.2.0  h1d223b6_15          conda-forge/linux-64      432 KB
  + liblapack              3.9.0  14_linux64_openblas  conda-forge/linux-64       12 KB
  + libnsl                 2.0.0  h7f98852_0           conda-forge/linux-64       31 KB
  + libopenblas           0.3.20  pthreads_h78a6416_0  conda-forge/linux-64       10 MB
  + libstdcxx-ng          11.2.0  he4da1e4_15          conda-forge/linux-64        4 MB
  + libuuid               2.32.1  h7f98852_1000        conda-forge/linux-64       28 KB
  + libzlib               1.2.11  h166bdaf_1014        conda-forge/linux-64       60 KB
  + ncurses                  6.3  h27087fc_1           conda-forge/linux-64     1002 KB
  + numpy                 1.22.3  py39h18676bf_2       conda-forge/linux-64        7 MB
  + openssl                3.0.2  h166bdaf_1           conda-forge/linux-64        3 MB
  + pandas                 1.2.5  py39hde0f152_0       conda-forge/linux-64       12 MB
  + pip                   22.0.4  pyhd8ed1ab_0         conda-forge/noarch          2 MB
  + pretty_html_table     0.9.11  pyhd8ed1ab_1         conda-forge/noarch          9 KB
  + python                3.9.12  h2660328_1_cpython   conda-forge/linux-64       27 MB
  + python-dateutil        2.8.2  pyhd8ed1ab_0         conda-forge/noarch        240 KB
  + python_abi               3.9  2_cp39               conda-forge/linux-64        4 KB
  + pytz                  2022.1  pyhd8ed1ab_0         conda-forge/noarch        242 KB
  + readline                 8.1  h46c0cb4_0           conda-forge/linux-64      295 KB
  + setuptools            59.8.0  py39hf3d152e_1       conda-forge/linux-64     1023 KB
  + six                   1.16.0  pyh6c4a22f_0         conda-forge/noarch         14 KB
  + sqlite                3.38.2  h4ff8645_0           conda-forge/linux-64        2 MB
  + tk                    8.6.12  h27826a3_0           conda-forge/linux-64        3 MB
  + tzdata                 2022a  h191b570_0           conda-forge/noarch        121 KB
  + wheel                 0.37.1  pyhd8ed1ab_0         conda-forge/noarch         31 KB
  + xz                     5.2.5  h516909a_1           conda-forge/linux-64      343 KB
  + zlib                  1.2.11  h166bdaf_1014        conda-forge/linux-64       88 KB

  Summary:

  Install: 37 packages

  Total download: 77 MB

─────────────────────────────────────────────────────────────────────────────────────────

error    Could not write to file /tmp/conda/349dcbf7-2a57-4fbb-9e11-ef6d4470f655/pandas-1.2.5-py39hde0f152_0.tar.bz2: No space left on device
critical Write failed
critical Write failed
error    Could not write PID to lockfile (No space left on device)
warning  Cannot lock '/tmp/conda/349dcbf7-2a57-4fbb-9e11-ef6d4470f655/_openmp_mutex-4.5-1_gnu.tar.bz2'
    Waiting for other mamba process to finish
error    Could not write PID to lockfile (No space left on device)
error    LockFile can't be set at '/tmp/conda/349dcbf7-2a57-4fbb-9e11-ef6d4470f655/_openmp_mutex-4.5-1_gnu.tar.bz2'
    This could be fixed by changing the locks' timeout or cleaning your environment from previous runs
warning  LockFile creation for path '/tmp/conda/349dcbf7-2a57-4fbb-9e11-ef6d4470f655/_openmp_mutex-4.5-1_gnu.tar.bz2' failed, continuing without it
critical Write failed
critical Write failed
```

Reply:
Mhm, still /tmp is inside the container. Maybe singularity maps it to some host dir implicitly, and thereby does not respect TMPDIR by default?

Reply:
Try to set SINGULARITY_TMPDIR to the place on scratch where you expect enough space.

Reply:
Or try setting SINGULARITY_LOCALCACHEDIR

Reply:
or both

Reply:
But the /tmp/conda folder is located within the container or do you see its content on the host outside of the container?

Reply:
I still don't get it, sorry. You are using the `container`/`singularity` directive though? The change you made in this PR will only affect the conda behavior **within** a singularity container invoked by Snakemake via the  `container`/`singularity` directive. I expected /tmp to be a folder within the container, that is mapped to some cache managed by singularity. I was expecting this cache could be configured via a variable like SINGULARITY_LOCALCACHEDIR.

Since the path defined in the statement you changed is however local to the container (and yes, somehow backed by the system but still local in the container) it does not make sense to put the system TMPDIR variable there, because that directory might not at all be visible in the container. Am I wrong?

Reply:
Let me ask postmaster if `MOUNT TMP` is set, that would explain this behaviour

> MOUNT TMP: When this option is enabled, Singularity will automatically bind mount /tmp and /var/tmp into the container from the host. If the --contain option is passed, Singularity will create both locations within the sessiondir or within the directory specified by the --workdir option if that is passed as well.

Reply:
Ok so `MOUNT TMP` is indeed set. I guess that means this PR can be closed?

Reply:
Would you mind adding a sentence about this to the docs, e.g. in the FAQ?